File: backend/BookstoreTamAn/src/main/java/com/minhnghia/datn/BookstoreTamAn/model/Order.java
package com.minhnghia.datn.BookstoreTamAn.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Order extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "order_code", nullable = false, unique = true)
    private String orderCode;
    private LocalDate orderDate;
    private String status;
    private Double totalPrice;
    private String email;
    private String paymentMethod;
    private String phone;
    private String shippingAddress;
    private String recipientName;
    private boolean active;
    private String createdBy;
    private String modifyBy;
    private Boolean paid;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<OrderDetail> orderDetails;
}

File: frontend/frontend/src/app/models/order.ts
export class Order{
    orderId: 0;
    orderCode : string;
    userId : 0;
    orderDate : string;
    status : string;
    totalPrice : 0;
    email : string;
    paymentMethod : string;
    phone : string;
    shippingAddress : string;
    recipientName : string;
    active : boolean;
    paid : boolean;
    constructor(){
        this.orderId = 0;
        this.orderCode = "";
        this.userId = 0;
        this.orderDate = "";
        this.status = "";
        this.totalPrice = 0;
        this.email = "";
        this.paymentMethod = "";
        this.phone = "";
        this.shippingAddress = "";
        this.recipientName = "";
        this.active = true;
        this.paid = false;
    }
}

export interface IListOrder{
    orderId: 0;
    orderCode : string;
    userId : 0;
    orderDate : string;
    status : string;
    totalPrice : 0;
    email : string;
    paymentMethod : string;
    phone : string;
    shippingAddress : string;
    recipientName : string;
    active : boolean;
    paid : boolean;
}